File: src/BinarySearch/FindPeakEle.java
package BinarySearch;

public class FindPeakEle {
    public static int findPeakElement(int[] nums) {
        int s = 0 ;
        int e = nums.length-1;
        while(s<=e){
            if (s==e) return s;
            int mid = s+(e-s)/2;
            if(nums[mid] <nums[mid+1]) s=mid+1;
            else e = mid;
        }return -1;
    }

    static void main(String[] args) {
        int [] arr = {1,2,1,3,5,6,4};
        System.out.print(findPeakElement(arr));
    }
}

File: src/CPP/BinarySearch/FindPeakEle.cpp
#include <iostream>
#include <vector>

using namespace std ;
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int s = 0 ;
        int e = nums.size()-1;
        while (s<=e){
            int mid = s +(e-s)/2;
            if (s==e) return s;
            else if (nums[mid] <nums[mid+1]) s= mid+1;
            else e= mid ;
        }return -1;
    }
};